File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:

        if x < 0 :
            return False
        
        original_x = x
        reverse_num = 0
        
        while x > 0:

            reverse_num = reverse_num * 10 + (x % 10)
            x = x // 10
            
        return reverse_num == original_x

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int num = 0;
        int originalnum = x;
        if(x < 0) {
        	return false;
        }
        else {
        	while(x >= 10) {
        		num =(num * 10) + (x % 10);
        		x = x / 10;
        	}
        	num = (num * 10) + x;
        	return num == originalnum;
        }
    }
}